mlflow/recipes/recipe.py

```diff
@@ -94,18 +94,19 @@ def run(self, step: str = None) -> None:
             last_executed_step_output_directory
         )
         if last_executed_step_state.status != StepStatus.SUCCEEDED:
+            last_step_error_mesg = (
+                f"The following error occurred while running step '{last_executed_step}':\n"
```

Comment:
```suggestion
                f"The following error occurred while running step '{last_executed_step.name if last_executed_step is not None else None}':\n"
```

Reply:
Because step class does not override `__str__` or `__repr__`, directly printing it will get `XXStep object at XXX` string

Reply:
That's a good point. I added a simple __str__ for BaseStep.